### ISSUE TITLE
na_ontap_broadcast_domain_ports module falsely supports check mode

### ISSUE THREAD

**Issue by stobias123** [ansible-collections/netapp#36]
##### SUMMARY
Edit:  Further investigation shows that this might be a broken module.

Running in **check mode** shows "0 changes" in this task...
```
TASK [na_ontap_cluster_config : Remove ports from Default broadcast domain] ****
```
and "1 change" in this task...
```
TASK [na_ontap_cluster_config : create broadcast domain] ***********************
changed: [localhost] => (item={u'mtu': 1500, u'ipspace': u'Default', u'ports': u'cluster01-01:e0M,cluster01-01:e0i,cluster01-02:e0M,cluster01-02:e0i', u'name': u'Default'})
````

When the playbook runs, I end up seeing the below.
```
TASK [na_ontap_cluster_config : Remove ports from Default broadcast domain] ****
failed: [localhost] (item={u'node': u'cluster01-01', u'autonegotiate': False, u'flowcontrol': u'none', u'port': u'e0M', u'mtu': 1500}) => {"ansible_loop_var": "item", "changed": false, "item": {"autonegotiate": false, "flowcontrol": "none", "mtu": 1500, "node": "cluster01-01", "port": "e0M"}, "msg": "Error deleting port for broadcast domain Default: NetApp API failed. Reason - 13001:Port \"cluster01-01:e0M\" 
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NaApiError: NetApp API failed. Reason - 13001:Port "cluster01-02:e0M" cannot be used because it is currently the home port or current port of a LIF. To view LIFs with a specified home port, use the "network interface show -home-node <nodename> -home-port {<netport>|<ifgrp>}" command. To view LIFs with a specified current port, use the "network interface show -curr-node <nodename> -curr-port {<netport>|<ifgrp>}" command.
failed: [localhost] (item={u'node': u'cluster01-02', u'autonegotiate': False, u'flowcontrol': u'none', u'port': u'e0M', u'mtu': 1500}) => {"ansible_loop_var": "item", "changed": false, "item": {"autonegotiate": false, "flowcontrol": "none", "mtu": 1500, "node": "cluster01-02", "port": "e0M"}, "msg": "Error deleting port for broadcast domain Default: NetApp API failed. Reason - 13001:Port \"cluster01-02:e0M\" cannot be used because it is currently the home port or curr…
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`ontap_cluster_config` role.

`na_ontap_broadcast_domain_ports` module.

##### STEPS TO REPRODUCE
Try to modify a broadcast domain with the provided role.

##### EXPECTED RESULTS
I expect the playbook check mode to reflect behavior of a run.

##### ACTUAL RESULTS
 Check mode shows nothing should be happening in the step where I see a failure.

**Comment by stobias123** [comment]
Looks like the module isn't actually comparing anything in check mode? 😱 

https://github.com/ansible-collections/netapp/blob/master/ansible_collections/netapp/ontap/plugins/modules/na_ontap_broadcast_domain_ports.py#L191

**Comment by carchi8py** [comment]
Not sure if you're in the Slack channel. We are aware of this, and have internal stories for each of the modules that should be supporting Check Mode but aren't, and should hopefully have the fixes out for the May Collection update.

**Comment by lonico** [comment]
It seems we missed this one.  We'll track it internally with DEVOPS-3346.

**Comment by lonico** [comment]
fixed with 195ce10a60e1e4f92afad09a0cc6c12cf0fb88ab

This will be released in Galaxy as 20.12.0 early December.
